File: src/main/kotlin/io/github/jonathanpotts/cartographytable/models/ChunkModel.kt
package io.github.jonathanpotts.cartographytable.models

import kotlinx.serialization.Serializable

/**
 * Stores data used to process a chunk.
 *
 * @property blocks A map of coordinates to block models (YXZ format).
 */
@Serializable
data class ChunkModel(
    var blocks: MutableMap<Short, Map<Byte, Map<Byte, BlockModel>>> =
        mutableMapOf()
)

File: web/src/models/ChunkModel.ts
import BlockModel from './BlockModel';

/**
 * Stores data used to process a chunk.
 */
interface ChunkModel {
  /**
   * A record of coordinates to block models (YXZ format).
   */
  blocks: Record<number, Record<number, Record<number, BlockModel>>>;
}

export default ChunkModel;
